game/src/main/scala/io/aigar/game/Resources.scala

```diff
@@ -24,74 +27,97 @@ object Silver {
 object Gold {
   final val Max = 10
   final val Min = 5
+  final val RespawnRetryAttempts = 10
 
   // IMPORTANT: keep those values in sync with the client documentation
   final val Mass = 0
   final val Score = 10
 }
 
-class Resources(grid: Grid) {
-  val regular = new ResourceType(grid, Regular.Min, Regular.Max, Regular.Mass, Regular.Score)
-  val silver = new ResourceType(grid, Silver.Min, Silver.Max, Silver.Mass, Silver.Score)
-  val gold = new ResourceType(grid, Gold.Min, Gold.Max, Gold.Mass, Gold.Score)
-  var resourceTypes = List(regular, silver, gold)
+class Resources(grid: Grid) extends EntityContainer {
+  var regulars = List.fill(Regular.Max)(new Regular(grid.randomPosition))
```

Comment:
Could we keep something similar to `ResourceType` (which would be an `EntityContainer`)? Because now it feels like we're back to duplicating the logic for the 3 resource types through copy-paste... Or would you have another idea?